src/Backups/BackupCoordinationLocal.cpp

```diff
@@ -33,42 +33,42 @@ Strings BackupCoordinationLocal::waitForStage(const String &, std::chrono::milli
     return {};
 }
 
-void BackupCoordinationLocal::addReplicatedPartNames(const String & table_shared_id, const String & table_name_for_logs, const String & replica_name, const std::vector<PartNameAndChecksum> & part_names_and_checksums)
```

Comment:
The main problem with the earlier solution was that `table_shared_id` was used to identify sets of parts of a replicated table. That was not correct in case of using multiple shards - because parts on different shards are likely to be different. I've changed that solution - now a zookeeper path is used instead of `table_shared_id`. The zookeeper path is the same as it's set in the first argument of the create query which created a replicated table (i.e. it contains the shard name but doesn't the replica name)